### PR TITLE
A few README tweaks

### PR DESCRIPTION
**Pull request by hzsweers**
- Android configuration
- Fixed some naming
- Make the java example use the more modern `java-library`
- Make jsr305 `compileOnly`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -25,32 +25,40 @@ buildscript {
 plugins {
   id "net.ltgt.apt" version "0.11"
   id "net.ltgt.errorprone" version "0.0.11"
-  id "java"
+  id "java-library"
```
Comment (kageiit) [inline README.md]: what about folks who use the standard "java" plugin? This might be confusing for that use case right?

Comment (kageiit) [reply README.md]: We should probably cover both otherwise it might give the wrong message that nullaway is not supported with the older "java" plugin

Comment (hzsweers) [reply README.md]: I can add a comment, I suppose `java` is still used for some applications

Comment (hzsweers) [reply README.md]: 5217d2f

File: README.md [hunk 2]
```diff
@@ -25,32 +25,40 @@ buildscript {
...
+    // Optional, some source of nullability annotations
+    compileOnly "com.google.code.findbugs:jsr305:3.0.2"
 
     errorprone "com.google.errorprone:error_prone_core:2.1.1"
     errorprone "org.checkerframework:dataflow:2.1.14"
-
 }
 
-compileJava {
+tasks.withType(JavaCompile) {
```
Comment (kageiit) [inline README.md]: This applies the same checks to test code as well. We should mention that and provide a way for folks to only apply to main code (as it is now) as test code may not be as important for null warnings

File: README.md [hunk 3]
```diff
@@ -25,32 +25,40 @@ buildscript {
...
-  id "java"
+  id "java-library" // Or just plain old "java"
 }
 
 dependencies {
     apt "com.uber.nullaway:nullaway:0.1.2"
 
-    compile "com.google.code.findbugs:jsr305:3.0.2"
+    // Optional, some source of nullability annotations
+    compileOnly "com.google.code.findbugs:jsr305:3.0.2"
```
Comment (kageiit) [inline README.md]: CompileOnly is not compatible with the `tasks.withType(JavaCompile)` bit as testCompileOnly does not extend from compileonly

Comment (hzsweers) [reply README.md]: What do you suggest instead? I don't think recommending people compile this is the right approach

File: README.md (hunk 2 above)
Comment (hzsweers) [reply README.md]: I can add a bit clarifying. I don't know off hand how this can be made to exclude test code though, do you?

Comment (msridhar) [reply README.md]: FWIW I think our default configuration should not run NullAway on test code; the payoff there is less clear.  The point of this change was to unify Java and Android projects, but this PR already has a small separate section for Android.  Couldn't we just make Android different in this respect as well?

File: README.md (hunk 3 above)
Comment (msridhar) [reply README.md]: Here I think we are fine assuming we don't suggest running NullAway on test code by default.

Comment (kageiit) [reply README.md]: `compileOnly` is fine here

File: README.md (hunk 2 above)
Comment (kageiit) [reply README.md]: I think we should leave this line unchanged as that indicates we are just talking about main code

File: README.md [hunk 4]
```diff
@@ -25,32 +25,40 @@ buildscript {
 plugins {
   id "net.ltgt.apt" version "0.11"
   id "net.ltgt.errorprone" version "0.0.11"
-  id "java"
+  id "java-library" // Or just plain old "java"
```
Comment (kageiit) [inline README.md]: the conventions with `java-library` are no longer `compile` etc. This configuration as is will show warnings to the user to switch to `api` and `implementation`. We should be consistent with the example code.

We should either use all the correct conventions of `java-library` or `java` (in this case, `annotationProcessor` is not a valid configuration in the old plugin). So I think it is better to just have two separate examples. One for java and one for android

**Comment by msridhar** [comment]
Made another pass.  @kageiit PTAL

**Pull request merged**